compiler/luci/service/src/Nodes/CircleStridedSlice.test.cpp

```diff
@@ -41,3 +42,351 @@ TEST(CloneNodeTest, clone_StridedSlice)
+  end.shape_status(luci::ShapeStatus::VALID);
+
+  strides.dtype(loco::DataType::S32);
+  strides.shape({4});
+  strides.size<loco::DataType::S32>(4);
+  strides.at<loco::DataType::S32>(0) = 1;
+  strides.at<loco::DataType::S32>(1) = 1;
+  strides.at<loco::DataType::S32>(2) = 1;
+  strides.at<loco::DataType::S32>(3) = 1;
+  strides.shape_status(luci::ShapeStatus::VALID);
+
+  strided_slice.input(&input);
+  strided_slice.begin(&begin);
+  strided_slice.end(&end);
+  strided_slice.strides(&strides);
+
+  loco::TensorShape shape;
+  luci::sinf::Rule shape_inf_rule;
+
+  ASSERT_TRUE(shape_inf_rule.infer(&strided_slice, shape));
+  ASSERT_EQ(4, shape.rank());
+  ASSERT_FALSE(shape.dim(0).known());
+  ASSERT_TRUE(shape.dim(1).known());
+  ASSERT_TRUE(shape.dim(2).known());
+  ASSERT_TRUE(shape.dim(3).known());
+  ASSERT_EQ(0, shape.dim(0).value());
+  ASSERT_EQ(3, shape.dim(1).value());
+  ASSERT_EQ(5, shape.dim(2).value());
+  ASSERT_EQ(7, shape.dim(3).value());
+}
+
+TEST(ShapeRuleTest, strided_slice_dynamic_shape_2)
+{
+  luci::CircleInput input;
+  luci::CircleConst begin;
+  luci::CircleConst end;
+  luci::CircleConst strides;
+
+  luci::CircleStridedSlice strided_slice;
+
+  input.shape({5, 6, 7, 8});
+  input.shape_status(luci::ShapeStatus::VALID);
+  input.dim(0).unset();
+
+  begin.dtype(loco::DataType::S32);
+  begin.shape({4});
+  begin.size<loco::DataType::S32>(4);
+  begin.at<loco::DataType::S32>(0) = 0;
+  begin.at<loco::DataType::S32>(1) = 0;
+  begin.at<loco::DataType::S32>(2) = 0;
+  begin.at<loco::DataType::S32>(3) = 0;
+  begin.shape_status(luci::ShapeStatus::VALID);
+
+  end.dtype(loco::DataType::S32);
+  end.shape({4});
+  end.size<loco::DataType::S32>(4);
+  end.at<loco::DataType::S32>(0) = 1;
+  end.at<loco::DataType::S32>(1) = 3;
+  end.at<loco::DataType::S32>(2) = 5;
+  end.at<loco::DataType::S32>(3) = 7;
+  end.shape_status(luci::ShapeStatus::VALID);
+
+  strides.dtype(loco::DataType::S32);
+  strides.shape({4});
+  strides.size<loco::DataType::S32>(4);
+  strides.at<loco::DataType::S32>(0) = 1;
+  strides.at<loco::DataType::S32>(1) = 1;
+  strides.at<loco::DataType::S32>(2) = 1;
+  strides.at<loco::DataType::S32>(3) = 1;
+  strides.shape_status(luci::ShapeStatus::VALID);
+
+  strided_slice.input(&input);
+  strided_slice.begin(&begin);
+  strided_slice.end(&end);
+  strided_slice.strides(&strides);
+  strided_slice.new_axis_mask(1);
+
+  loco::TensorShape shape;
+  luci::sinf::Rule shape_inf_rule;
+
+  ASSERT_TRUE(shape_inf_rule.infer(&strided_slice, shape));
+  ASSERT_EQ(5, shape.rank());
+  ASSERT_TRUE(shape.dim(0).known());
+  ASSERT_FALSE(shape.dim(1).known());
+  ASSERT_TRUE(shape.dim(2).known());
+  ASSERT_TRUE(shape.dim(3).known());
+  ASSERT_TRUE(shape.dim(4).known());
+  ASSERT_EQ(1, shape.dim(0).value());
+  ASSERT_EQ(0, shape.dim(1).value());
+  ASSERT_EQ(5, shape.dim(2).value());
+  ASSERT_EQ(7, shape.dim(3).value());
+  ASSERT_EQ(8, shape.dim(4).value());
+}
+
+TEST(ShapeRuleTest, strided_slice_NEG_1)
```

Comment:
We could give more specific names to the tests, I think.

Reply:
Thank you, I will rename it a more specific.